File: 2_Periodo/AED_II/TPS/Arquivo.java
import java.io.RandomAccessFile;
import java.util.*;
/**
 * Arquivo
 * @author Vinicius Francisco da Silva
 * @version 1 08/2017
 */
public class Arquivo{
	public static void main(String[] args) throws Exception{
		String str;
		RandomAccessFile randomaccessfile = new RandomAccessFile("outarq","rw");
		int n = MyIO.readInt();
		random(n,randomaccessfile);
	}// End main()

	/*
     * Metodo publico escreve n valores em um arquivo
     * fecha o mesmo e realiza a leitura do final do
     * arquivo até o inicio
     * @param int n com a quantidade de valores a serem
     * lido, RandomAccessFile com um ponteiro que está
     * apontando para o endereço onde está localizado o 
     * arquivo
     * @return void
     */
	public static void random(int n,RandomAccessFile randomaccessfile) throws Exception{
		for(int i = 0; i < n; i++){
			randomaccessfile.writeDouble(MyIO.readDouble());
		}// End for
		long j = randomaccessfile.getFilePointer();
		double dbl;
		randomaccessfile.close();
		randomaccessfile = new RandomAccessFile("outarq","r"); 
		j = j-8;
		while(j >= 0){
			randomaccessfile.seek(j);
			dbl = randomaccessfile.readDouble();
			if(dbl - (int)dbl == 0){
				MyIO.println((int)dbl);
			}// End if
			else{
				MyIO.println(dbl);
			}// End else
			j = j-8;
		}// End while
		randomaccessfile.close();
	}// End random()
}// End class Arquivo

File: 2_Periodo/AED_II/TPS/Arquivo.cpp
#include <stdio.h>
#include <stdlib.h>
/**
 * Arquivo em C
 * @author Vinicius Francisco da Silva
 * @version 1 08/2017
 */
void arquivo(int n,FILE* file);
int main(){
	int n;
	FILE* file;
	scanf("%i",&n);
	arquivo(n,file);
	return 0;
}// End main()

/*
     * função escreve n valores em um arquivo
     * fecha o mesmo e realiza a leitura do final do
     * arquivo até o inicio
     * @param int n com a quantidade de valores a serem
     * lido, ponteiro do tipo FILE que está
     * apontando para o endereço onde está localizado o 
     * arquivo
     * @return void
     */
void arquivo(int n,FILE* file){
	file = fopen("outarq","wb+");
	double dbe;
	double dbl;
	for(int i = 0;i < n;i++){
		scanf("%lf",&dbe);
		fwrite(&dbe,sizeof(double),1,file); 
	}// End for 
	fclose(file);
	file = fopen("outarq","rb+");
	n--;	
	while(n >= 0){
		fseek(file,n*sizeof(double),SEEK_SET);
		fread(&dbl,sizeof(double),1,file);
		if(dbl - (int)dbl == 0){
			printf("%i\n",(int)dbl);		
		}// End if
		else{
			printf("%g\n",dbl);
		}// End else
		n--;
	}// End while
	fclose(file);
}// end arquivo()